src/bp/ui-studio/src/web/views/FlowBuilder/diagram/nodes_v2/ActionNode.tsx

```diff
@@ -0,0 +1,142 @@
+import classnames from 'classnames'
+import { ActionServer } from 'common/typings'
+import _ from 'lodash'
+import React, { FC, useState } from 'react'
+import { AbstractNodeFactory, DiagramEngine } from 'storm-react-diagrams'
+
+import ActionDialog from '../../nodeProps/ActionDialog'
+import { BaseNodeModel } from '../nodes/BaseNodeModel'
+import { StandardPortWidget } from '../nodes/Ports'
+
+import style from './style.scss'
+import { showHeader } from './utils'
+
+const ActionInfo: FC<{ action: Action }> = props => {
+  const { action } = props
+
+  return (
+    <div>
+      {action.name} ({action.actionServerId})
+    </div>
+  )
+}
+
+const ActionNodeContent: FC<{ action: Action; cancel: () => void; onSave: (action: Action) => void }> = props => {
+  const [showDialog, setShowDialog] = useState(false)
+
+  const { action, cancel, onSave } = props
+  return (
+    <div className={style.content} onDoubleClick={() => setShowDialog(true)}>
+      {/* <Button onClick={() => setShowDialog(true)}>Edit</Button> */}
+      <ActionInfo action={action} />
+      <ActionDialog
+        name={action.name}
+        parameters={action.parameters}
+        actionServerId={action.actionServerId}
+        isOpen={showDialog}
+        onClose={() => {
+          setShowDialog(false)
+          cancel()
+        }}
+        onSave={action => {
+          setShowDialog(false)
+          onSave(action)
+        }}
+      />
+    </div>
+  )
+}
+
+const ActionWidget: FC<{
+  node: ActionNodeModel
+  diagramEngine: any
+}> = props => {
+  const { node, diagramEngine } = props
+
+  const parseActionString = (actionString: string | undefined): Action => {
+    let name = ''
+    let parameters = {}
+    let actionServerId = ''
+
+    if (actionString) {
+      const chunks = actionString.split(' ')
+      name = chunks[0]
+
+      const parametersString = chunks[1]
+      parameters = JSON.parse(parametersString)
+
+      actionServerId = chunks[2]
```

Comment:
Same here as previous comment, the server id must be second since you may have an unknown number of spaces. 
Destructuring is nice here:
```js

const [name, parameters, actionServer] = actionString.split(' ')
```